### DISCUSSION TITLE
How to toggle one touch actions in the web socket

### DISCUSSION THREAD

**Discussion by bonro001** [aqualinkd/AqualinkD#233] [Q&A] [answered]
Does anyone know of a way to toggle one touch actions using the web socket, or for that matter, toggle anything using the socket?

**Comment by sfeakes** [comment, marked as answer]
Aqualinkd does not support one touch buttons at the moment. It will in the future.  As for toggle anything else with web socket, it’s explained in the wiki.  (Or just put any browser in developer mode and look at how the web ui uses web sockets). But it’s not recommended to use web socket. Using MQTT or the web API is the best option as the web sockets are reserved for Aqualinkd’s web UI and can change between releases.

**Reply by bonro001** [reply]
Thank you.  I guess followup question, when turning on the spa do you know if I can reliably turn on spa mode, the heater and the pump at the same time?  Is there any concept in the API to turn on multiple items in a single call?

**Reply by sfeakes** [reply]
To turn on filter pump, spa and spa heater you will need to make 3 API calls.  This is also true using websockets.  AqualinkD will accept those 3 calls within milliseconds, so it’s not a problem for any caller to make them consecutively, Ie without any delay in code.

**Reply by bonro001** [reply]
Curious if this has been noticed, it seems that it is possible to get the spa mode setting backwards.  Not sure if it was just our testing or sending a command while the valves are turning, but the pool thinks they are in pool mode when they are in spa mode.  Only way to reset this is to reboot the pool box.

**Reply by sfeakes** [reply]
Never seen / had anyone mention that problem before.  If you have a PDA only panel, then I could see getting things backwards for 10 or so seconds within AqualinkD, (limitation of PDA panels) but the Control Panel would sort itself out (no need for a reboot).  If you are using a normal panel then setting spa mode would be almost instant in reporting the Valve Actuator is moving.   Did the actuator get stuck or something?
